src/Uno.UI/Controls/Window.macOS.cs

```diff
@@ -44,8 +44,210 @@ public Window(CGRect contentRect, NSWindowStyle aStyle, NSBackingStore buffering
 			Delegate = new WindowDelegate(this);
 			_inputPane = InputPane.GetForCurrentView();
 			_inputPane.Window = this;
+
+			// Register the types that can be dragging into the window (effectively 'AllowDrop').
+			// This must be done before dragging enters the window or dragging methods won't be invoked.
+			// We register for all possible types and then confirm whether they
+			// are actually supported within the draggingEntered and draggingUpdated methods.
+			// This has a minor side effect in the standard UI (zoom effect) but is considered acceptable.
```

Comment:
When you say it "has a minor effect in the standard UI", I was wondering if we should maintain a list of loaded elements that allow drop. If so, we could at least invoke this registration only if we have an element that allows drop (and even maybe unregister if element gets unloaded).

We would still not be able to pre-filter supported types like macOS seems to ask for, but we could at least not appear as a possible drop target if nothing is supported.

Does this would significantly improve the user experience?

Reply:
It is such a minor user facing issue. No plus icon appears next to the cursor at this point. Just the dragging image frame will zoom out slightly over an Uno App indicating it is a target. Without the plus appearing though it is clearly not supporting drop unless we accept it which is what users would expect. It is possible to maintain a global list of registered UIElement and their accepted drop types. However we would also need notification to every window when this global list changes. It seems like a lot of work that still wouldn't completely fix this. It might be better to wait and see if there are user complaints before attempting something here. I expect most will never notice.